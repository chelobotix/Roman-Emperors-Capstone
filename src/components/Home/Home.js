import { v4 as uuidv4 } from 'uuid';
import { useSelector } from 'react-redux';
import { useState } from 'react';
import DynastyCard from '../DynastyCard/DynastyCard';
import Heading from '../Heading/Heading';
import style from './Home.module.css';
import romanHelmet from '../../assets/images/romanHelmet.png';
import romanHelmet2 from '../../assets/images/romanHelmet2.png';
import romanHelmet3 from '../../assets/images/romanHelmet3.png';
import romanHelmet4 from '../../assets/images/romanHelmet4.png';

const Home = () => {
  const state = useSelector((state) => state.emperors);
  const [searchDynasty, setSearchDynasty] = useState([...state.dynasty]);
  const imgArray = [romanHelmet, romanHelmet2, romanHelmet3, romanHelmet4];
  let result = [];

  const handlerSearch = (e) => {
    if (e.target.value === '') {
      result = [...state.dynasty];
    } else {
      result = state.dynasty.filter((data) => (
        data.name.toLowerCase().includes(e.target.value.toLowerCase())
      ));
    }
    setSearchDynasty([...result]);
  };

  const listStyle = () => {
    let toggleColor = true;
    let counter = 1;
    let imgIndex = 0;
    return (
      <ul className={style.ul}>
        {searchDynasty.map((dynasty) => {
          counter += 1;
          if (counter === 3) {
            toggleColor = !toggleColor;
            counter = 1;
          }
          if (imgIndex === 3) {
            imgIndex = 0;
          } else {
            imgIndex += 1;
          }
          return (
            <li key={uuidv4()} className={toggleColor ? `${style.li} ${style.liOne}` : `${style.li} ${style.liTwo}`}>
              <DynastyCard name={dynasty.name} count={dynasty.count} img={imgArray[imgIndex]} />
            </li>
          );
        })}
      </ul>
    );
  };

  return (
    <>
      <Heading />
      <div className={style.mainContainer}>
        <div className={style.searchContainer}>
          <input type="text" onChange={handlerSearch} placeholder="Search By Dynasty..." className={style.search} />
        </div>
        <div className={style.dynastyContainer}>
          <h2 className={style.h2}>Dynasty</h2>
          {listStyle()}
        </div>
      </div>
    </>
  );
};

export default Home;
