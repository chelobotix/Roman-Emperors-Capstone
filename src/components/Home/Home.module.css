.mainContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: #fff;
}

.searchContainer {
  width: 100%;
  display: flex;
  justify-content: center;
  background-color: #3d60a2;
}

.search {
  width: 95%;
  max-width: 500px;
  height: 30px;
  margin: 0 5px 0 5px;
  border-radius: 10px;
  border: 1px solid #3d60a2;
  padding-left: 5px;
}

.dynastyContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
}

.ul {
  width: 100%;
  list-style-type: none;
  display: flex;
  flex-wrap: wrap;
}

.li {
  width: 50%;
  max-width: 500px;
  height: 200px;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.liOne {
  background-color: #4369b0;
}

.liTwo {
  background-color: #3d60a2;
}

.h2 {
  width: 100%;
  text-align: center;
  background-color: #4369b2;
  color: #fff;
  border-bottom: 1px solid #fff;
}
