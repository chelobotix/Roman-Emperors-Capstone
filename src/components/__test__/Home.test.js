import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';
import Home from '../Home/Home';

const dynasty = [
  {
    name: 'Augustus',
    mission_name: 'Mission test',
    description: 'Description Test',
    reserved: false,
  },
];

const mockStore = configureStore([]);
const initialState = {
  emperors: {
    dynasty,
  },
};

const store = mockStore(initialState);

describe('Home', () => {
  test('Check if render', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>,
    );
    const ul = screen.getByRole('list');
    expect(ul).toBeInTheDocument();
  });

  test('Home Tree', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
