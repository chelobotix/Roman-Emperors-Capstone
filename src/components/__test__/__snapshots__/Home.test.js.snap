// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Home Tree 1`] = `
Array [
  <div
    className="heading"
  />,
  <div
    className="mainContainer"
  >
    <div
      className="searchContainer"
    >
      <input
        className="search"
        onChange={[Function]}
        placeholder="Search By Dynasty..."
        type="text"
      />
    </div>
    <div
      className="dynastyContainer"
    >
      <h2
        className="h2"
      >
        Dynasty
      </h2>
      <ul
        className="ul"
      >
        <li
          className="li liOne"
        >
          <a
            className="link"
            href="/dynasty/Augustus"
            onClick={[Function]}
          >
            <img
              alt="enter"
              className="enter"
              src="enter.png"
            />
          </a>
          <img
            alt="Dynasty"
            className="img"
            src="romanHelmet2.png"
          />
          <h3
            className="h3"
          >
            Augustus
          </h3>
          <p
            className="p"
          />
        </li>
      </ul>
    </div>
  </div>,
]
`;
